src/primo/browzine-primo-adapter.js

```diff
@@ -247,21 +247,23 @@ browzine.primo = (function() {
         if(browzineWebLink && browzineEnabled) {
           var template = buildTemplate(scope, browzineWebLink);
           element.append(template);
-          $scope.$apply();
         }
 
         if(coverImageUrl && browzineEnabled) {
-          setTimeout(function() {
-            var elementParent = getElementParent(element);
-            var coverImages = elementParent.querySelectorAll("prm-search-result-thumbnail-container img");
-
-            Array.prototype.forEach.call(coverImages, function(coverImage) {
-              coverImage.src = coverImageUrl;
-              $scope.$apply();
+          (function poll(){
+            setTimeout(function(){
```

Comment:
If we aren't setting a timeout value here (a second parameter to the `setTimeout` function), I believe the value defaults to 0. Do we still need to wrap this function in a `setTimeout`? Or are we using it to emulate the behavior of `setImmediate`?

Reply:
> Do we still need to wrap this function in a setTimeout? 

Yes.

> Or are we using it to emulate the behavior of setImmediate?

Definitely the right ballpark here. Basically, the UI is updated asynchronously, so without the setTimeout, our code (which updates the UI), will run before the UI is ready. We don't actually need to wait any actual unit of time. What we really need to do is get our code in queue to execute after the display updates and any events ahead of us have completed.

For a very good article on microtask queueing and execution within the browser's event loop. See, https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/